api/dispatcher.proto

```diff
@@ -157,3 +165,55 @@ message TasksMessage {
 	repeated Task tasks = 1;
 }
 
+message AssignmentsRequest {
+	string session_id = 1 [(gogoproto.customname) = "SessionID"];
+}
+
+message AssignmentsMessage {
+	// AssignmentType specifies whether this assignment message carries
+	// the full state, or is an update to an existing state.
+	enum AssignmentType {
+		COMPLETE = 0;
+		INCREMENTAL = 1;
+	}
+
+	AssignmentType assignment_type = 1;
+
+	// Sequence is the sequence number of the assignment message. This
+	// starts at 0 and increments. If a sequence number is skipped, the
+	// consumer of the stream should start a new Assignments stream to
+	// re-sync.
+	int64 sequence = 2;
```

Comment:
Should there be two opaque strings - one identifying the current message and another to match with the previous message?